lib/smart_answer/state_resolver.rb

```diff
@@ -0,0 +1,85 @@
+module SmartAnswer
+  class StateResolver
+    def initialize(flow)
+      @flow = flow
+    end
+
+    def state_from_response_store(response_store, requested_node = nil)
+      forwarding_responses = @flow.response_store == :session ? {} : response_store.all
+      establish_response_store_state(start_state(forwarding_responses),
+                                     response_store,
+                                     requested_node)
+    end
+
+    def state_from_params(params)
+      responses = params[:responses].to_s.split("/")
+
+      state = responses.inject(start_state) do |current_state, response|
+        return current_state if current_state.error
+
+        transition_state_to_next_node(current_state, response)
+      end
+
+      if params[:next]
+        transition_state_to_next_node(state, params[:response])
+      elsif params[:previous_response]
+        apply_response_to_state(state, params[:previous_response])
+      else
+        state
+      end
+    end
+
+  private
+
+    def establish_response_store_state(state, response_store, requested_node)
```

Comment:
v minor nit: I couldn't really tell what this method is doing from this name. Maybe something like `resolve_state`. Again nothing major, as there's more confusing elements to this code. Happy to get the PR in then worry about this.

Reply:
Sure that seems reasonable "resolve" fits the existing vernacular of Resolver. I was a bit concerned that it benefits from having response_store in there to make it distinct from path based Smart Answers, but I guess the response_store argument is sufficient.